traitlets/traitlets.py

```diff
@@ -2477,8 +2477,9 @@ def instance_init(self, obj):
 class Dict(Instance):
     """An instance of a Python dict."""
     _trait = None
+    _key_trait = None
 
-    def __init__(self, trait=None, traits=None, default_value=Undefined,
+    def __init__(self, trait=None, key_trait=None, traits=None, default_value=Undefined,
```

Comment:
I don't see how there is a general one positional arg that would have a well defined meaning for all TraitType's, so the positional args could equally well be trait specific. But then, I guess backwards compatibility shouldn't be broken lightly, so it's anyway too late. I'll shut up now and will just fix backwards compatibility.